### ISSUE TITLE
Search in Resources NPE

### ISSUE THREAD

**Issue by SuperRiderTH** [IsmAvatar/LateralGM#372]
Attempting to type just "/", or "\\" in the search box makes it very upset and yell at you.
```
LateralGM Version: 1.8.40

Operating System: Linux
Version: 4.13.0-37-generic
Architecture: amd64

Java Name: OpenJDK 64-Bit Server VM
Java Vendor: Oracle Corporation
Version: 1.8.0_151

Available processors (cores): 8
Free memory (bytes): 263500864
Maximum memory (bytes): 1828716544
Total memory available to JVM (bytes): 849346560

File system root: /
Total space (bytes): 59064549376
Free space (bytes): 10405466112
Usable space (bytes): 7374647296

Stack trace:
java.lang.NullPointerException
	at org.lateralgm.main.Search.searchInResourcesRecursion(Search.java:590)
	at org.lateralgm.main.Search.searchInResourcesRecursion(Search.java:464)
	at org.lateralgm.main.Search.searchInResources(Search.java:919)
	at org.lateralgm.main.Search$11.actionPerformed(Search.java:1431)
	at javax.swing.JTextField.fireActionPerformed(JTextField.java:508)
	at javax.swing.JTextField.postActionEvent(JTextField.java:721)
	at javax.swing.JTextField$NotifyAction.actionPerformed(JTextField.java:836)
	at javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1663)
	at javax.swing.JComponent.processKeyBinding(JComponent.java:2882)
	at javax.swing.JComponent.processKeyBindings(JComponent.java:2929)
	at javax.swing.JComponent.processKeyEvent(JComponent.java:2845)
	at java.awt.Component.processEvent(Component.java:6310)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:806)
	at java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1074)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:945)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:771)
	at java.awt.Component.dispatchEventImpl(Component.java:4760)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at org.GNOME.Accessibility.AtkWrapper$5.dispatchEvent(AtkWrapper.java:700)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by RobertBColton** [comment]
Error happened here:
https://github.com/IsmAvatar/LateralGM/blob/51ba5fe8b146d51b12d54ba0208c1e65e6c0d54f/org/lateralgm/main/Search.java#L590

I am not sure if it could be related to OpenJDK, I have no problems entering either of those path separators in the search filter on Windows. You could give the Oracle JDK a try. Also, just fyi, I am focusing all of my efforts on a new IDE, I just can't maintain LGM anymore primarily because of its architecture and the abundance of Java bugs in the OpenJDK.

**Comment by RobertBColton** [comment]
I need exact steps to reproduce this before I will be able to fix it.

**Comment by RobertBColton** [comment]
Ok, I have begun finding a way to reproduce some exceptions with the search feature. Specifically, with searching a word that happens to be part of the name of a resource I just deleted. This was giving me some issues earlier. That may be a step towards resolving this problem.

**Comment by RobertBColton** [comment]
I have steps to reproduce it reliably now and am currently working on a fix for it.

1) Open LateralGM & create a new font.
2) Create a new object.
3) Add a draw set font action to the event.
4) Set the font argument to the font.
5) Save and close the font action.
6) Delete the font.
7) Search for anything in the search bar.

**Comment by RobertBColton** [comment]
Closing as resolved by 73c9acd69be4bf3ba5322e1d31585f67e59379a1 which cleans up the searching in actions logic. It does seem as though I've fixed the original issue here as my above steps are not reproducible now. I hope this helps!
